var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -221,7 +221,9 @@ class Llvm(CMakePackage, CudaPackage):
     depends_on("libxml2")
 
     # lldb dependencies
-    depends_on("swig", when="+lldb")
+    depends_on("swig", when="@:9 +lldb")
+    depends_on("swig@2:", when="@10:11 +lldb")
+    depends_on("swig@3:", when="@12: +lldb")
```

Comment:
```suggestion
    with when("+lldb"):
        depends_on("swig")
        depends_on("swig@2:", when="@10:")
        depends_on("swig@3:", when="@12:")
```

maybe a bit more concise, note that it's ok to have overlapping ranges in `when`, the effective constraint is the intersection